### ISSUE TITLE
Feature request. Expose the w3mmd variable

### ISSUE THREAD

**Issue by Stannnnn** [PBug90/w3gjs#30]
Hello again, can you also expose the w3mmd variable in the finalize function? I wanna get some custom data from a custom game. :D

Not exactly sure how w3mmd works tho from what I've parsed I've always gotten this type:

type W3mmdAction = {
    actionId: number;
    filename: string;
    missionKey: string;
    key: string;
    value: number;
};

Maybe set w3mmd's type to W3mmdAction[]

I'm writing my own 'parser' for this now cuz I can't really find anything public. If someone knows of something please let me know (:

**Comment by PBug90** [comment]
This parser supports W3MMD out of the box. If for some reason it does not work, please provide a replay.

All you have to do is listen for actionblocks like this:
```
parser.on('actionblock', (block, playerid) => {
   if (block.actionId === 0x6b){
       console.log(block)
   }
})
```

If you want to read more about W3MMD then check out its spec here: https://github.com/PBug90/w3gPlus/blob/master/W3MMD.spec

**Comment by Stannnnn** [comment]
Thanks! I guess there is no default block parser for W3MMD?

**Comment by PBug90** [comment]
Currently there isnt. What would a default block parser look like for you?

**Comment by Stannnnn** [comment]
Well w3mmd has a pretty limited amount of actions so a typed object as output holding the specific player vars and separate events should do it mostly. I'm still looking into this however I've stumbled upon a problem.

I'm parsing the following replay:
https://wc3stats.com/games/8898/mmd
As u can see on the site it says Drake53 has a score of 500

However when I parse the replay with this parser then I'm getting an output of 400. I've concluded that it's somehow missing actions..

When I store all the actions with id 0x6b in an array and log them I'll get:
5x VarP 0 score += .....
5x VarP 1 score += .....
6x VarP 2 score += .....
6x VarP 4 score += .....

The scores of players 2 & 4 are correct however 0 & 1 are somehow missing 1 action which is causing the score to be wrong.

Am I missing something in the implementation or is there an actual way for the parser to lose track of some actions?

**Comment by PBug90** [comment]
That is an interesting find. I will try to take a look at it today or at the weekend and come back to you with an explanation. 

You did not include any error logs, so I assume the parser managed to parse all command blocks successfully and did not enter the following line: 
https://github.com/PBug90/w3gjs/blob/ef22c0cecc13528529774797f9f29fbfa0a2c0f6/src/W3GReplay.ts#L154

Did you try to compare with the actual ingame values if you use wc3 to run the replay and compare the final values?

**Comment by Stannnnn** [comment]
Ah right.. the replay has about 20k unknown actions which was causing the parser to slow down enormously. (See https://github.com/PBug90/w3gjs/issues/33). So I commented out this line. I will recheck it later today or tomorrow.

The ingame values match the expected ones as wc3stats concludes so something here is going wrong.

**Comment by Stannnnn** [comment]
Hmm, I just rechecked it. (:
So I enabled the error logging again. Then I added the unknown action to actions.ts
0x7a: new Parser().skip(20)

When I parse a replay now it seems to have correctly parsed all the actions and the final results are the same as on wc3stats.

**Comment by PBug90** [comment]
I have just published 1.5.0 with a skip-parser for 0x7a, so parsing should work as you mentioned. Would you mind opening and working on a PR that implements a suitable data structure for W3MMD data?

**Comment by PBug90** [comment]
Will be tracked in #59

**Comment by Stannnnn** [comment]
Yo, I just tried your latest release however it looks like the W3MMDAction is never emitted. 

  replayParser.on('gamedatablock', (action: Action) => {
                    if (action.id === 0x6b) {
                        actionBlocks.push({ action, time: replayParser.msElapsed })
                    }
                })

My actionBlocks array stays empty, if I read the replayParser.w3mmd I do get all events but I need the time as well.

Am I doing something wrong again? :D

Also I got a very basic implementation for parsing the MMD events, could share the code if you're still interested since you're already working on something.
